runtime/src/bank.rs

```diff
@@ -3786,7 +3786,8 @@ impl Bank {
                     Some(index)
                 }
                 Err(TransactionError::WouldExceedMaxBlockCostLimit)
-                | Err(TransactionError::WouldExceedMaxAccountCostLimit) => Some(index),
+                | Err(TransactionError::WouldExceedMaxAccountCostLimit)
+                | Err(TransactionError::WouldExceedMaxAccountDataCostLimit) => Some(index),
```

Comment:
Do we need a corresponding line in `unlock_accounts` like we have for `WouldExceedMaxBlockCostLimit` and `WouldExceedMaxAccountCostLimit`?

Reply:
@taozhu-chicago

Reply:
yes, @jeffwashington, please add `TransactionError::WouldExceedMaxAccountDataCostLimit` to runtime/src/accounts.rs `unlock_accounts()` match cases, otherwise it'd unintentional unlock accounts that are not locked at first place.

Reply:
@taozhu-chicago please review the change I just pushed

Reply:
lgtm